File: lab09_Mosque_Management/untitled/src/CurrentTimeWidget.java
import java.util.Map;
import java.time.LocalTime;

class CurrentTimeWidget extends Component {
    private LocalTime currentTime;

    public void setCurrentTime(LocalTime currentTime) {
        this.currentTime = currentTime;
        mediator.notify(this, "timeUpdated");
    }

    public LocalTime getCurrentTime() {
        return currentTime;
    }

    @Override
    public void update() {
        System.out.println("Current Time Widget updated: " + currentTime);
    }
}

File: Lab9_Mosque_Management_System/Lab9_Mosque_Management_System/Widgets/CurrentTimeWidget.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab9_Mosque_Management_System
{
    public class CurrentTimeWidget : IWidget
    {
        private IMediator _mediator;

        public void SetMediator(IMediator mediator)
        {
            _mediator = mediator;
        }

        public void Update()
        {
            Console.WriteLine("Current time updated based on the current time zone");
            _mediator.Notify(this, "CurrentTimeUpdated");
        }
    }
}
